hs_core/models.py

```diff
@@ -1509,33 +1513,49 @@ class Meta:
 
 
 def get_path(instance, filename):
+    """
+    Dynamically determine storage path for a FileField based upon whether resource is federated
+
+    :param instance: instance of ResourceFile containing the FileField;
+    :param filename: the filename to be used; derived from upload data
+
+    The instance points to the Resource record, which contains the federation path.
+    """
+    # retrieve federation path -- if any -- from Resource object containing FileField
     if instance.content_object.resource_federation_path:
-        return os.path.join(instance.content_object.resource_federation_path,
-                            instance.content_object.short_id, 'data',
-                            'contents', filename)
+        # federated iRODS storage
+        prefix = os.path.join(instance.content_object.resource_federation_path,
+                              instance.content_object.short_id, 'data', 'contents')
+    else:
+        # local iRODS storage
+        prefix = os.path.join(instance.content_object.short_id, 'data', 'contents')
+
+    if instance.file_folder is not None:
+        # use subfolder
+        return os.path.join(prefix, instance.file_folder, filename)
     else:
-        return os.path.join(instance.content_object.short_id, 'data', 'contents', filename)
+        # use root folder
+        return os.path.join(prefix, filename)
 
 
 class ResourceFile(models.Model):
     object_id = models.PositiveIntegerField()
     content_type = models.ForeignKey(ContentType)
-
     content_object = GenericForeignKey('content_type', 'object_id')
-    resource_file = models.FileField(upload_to=get_path, max_length=500, null=True, blank=True,
-                                     storage=IrodsStorage()
-                                     if getattr(settings, 'USE_IRODS', False)
-                                     else DefaultStorage())
+    file_folder = models.CharField(max_length=255, null=True)
```

Comment:
@alvacouch It may be better to rename ```file_folder``` to ```file_upload_to_folder``` to make its role clearer, or add comment to indicate what it is used for just for easier reference.